docs/next/guides/building-and-testing/plugins.md

```diff
@@ -17,228 +15,242 @@ tags:
 
 ## Overview
 
-Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of the features available in this library are provided by plugins.
+Plugin is a one great way to extend the capabilities of Handsontable. In fact, most of features available in this library are provided by plugins.
 
-## Plugin template
+This guide will lead you, step-by-step, through the process of creating a custom extension.
 
-There are two types of plugins: internal and external. While both extend Handsontable's functionality, the former is incorporated into the Handsontable build, and the latter needs to be included from a separate file.
+### 1. Prerequisites
 
-Regardless of which plugin you are going to build, using a template will save you lots of time.
+You need to import the following:
+- `BasePlugin` - contains the basic interface to work within Handsontable's lifecycle,
+- `registerPlugin` - utility to register the plugin in Handsontable plugins registry.
 
-### Internal plugin
+Both provide the basic interface necessary to create a reusable plugin.
+
+```js
+import { BasePlugin, registerPlugin } from 'handsontable/plugins';
+```
+
+### 2. Extend BasePlugin
+Extending our `BasePlugin` is a recommended way to start creating your next extension.
+In this built-in interface, we care about many factors such as backward compatibility, memory leaks preventions and proper binding plugin's instance with Handsontable.
```

Comment:
~Both provide the basic interface necessary to create a reusable plugin.~

```js
import { BasePlugin, registerPlugin } from 'handsontable/plugins';
```

\### 2. Extend the \`BasePlugin\`
The best way to start creating your own plugin is to extend the \`BasePlugin\`.

The \`BasePlugin\` interface takes care of:
\* Backward compatibility
\* Memory leak prevention
\* Properly binding your plugin's instance to Handsontable